extensions/2.0/Khronos/KHR_texture_basisu/README.md

```diff
@@ -0,0 +1,154 @@
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_basisu": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.png"
+        },
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+When used in the glTF Binary (GLB) format the `image` that points to the KTX2 resource uses the `mimeType` value of `image/ktx2`.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_basisu"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_basisu": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX2 image with Basis Universal supercompression without a fallback, define `KHR_texture_basisu` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_basisu"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_basisu"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_basisu": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_basisu.schema.json](schema/texture.KHR_texture_basisu.schema.json)
+
+## Restrictions on KTX2 Images with Basis Universal Supercompression
+
+When used with glTF 2.0, KTX2 images with Basis Universal supercompression must conform to these additional restrictions:
+
+- Swizzling metadata (`KTXswizzle`) must be `rgba` or omitted.
+- Orientation metadata (`KTXorientation`) must be `rd` or omitted.
+- Colorspace information in the DFD must match the expected usage, namely:
+  - Color primaries must be set to `KHR_DF_PRIMARIES_BT709`.
+  - Transfer function must be either `KHR_DF_TRANSFER_LINEAR` or `KHR_DF_TRANSFER_SRGB`.
+
+### Using KTX2 Images with Basis Universal Supercompression for Material Textures
+
+When a texture referencing a KTX2 image with Basis Universal supercompression is used for glTF 2.0 material maps (such as base color), the KTX2 image must be of **2D** type as defined in the KTX2 Specification, namely:
+
+- `pixelHeight` must be greater than 0.
+- `pixelDepth` must be 0.
+- `layerCount` must be 0.
+- `faceCount` must be 1.
+
+In general, a KTX2 image used for material texture should contain a full mip pyramid.
```

Comment:
This line is normative, correct? If so, perhaps phrase this and the line below as:

A KTX2 image must contain a full mip pyramid, unless another extension depending on KHR_texture_basisu imposes different requirements.

Reply:
Do e.g. 2x2 and 1x1 levels bring so much value that the spec needs to _require_ them? I always thought that authors are free to decide where to stop with low-resolution levels.

Reply:
How does WebGL 1.0 handle texture completeness requirements? I'm struggling to find this in the WebGL 1 spec - however GLES2 spec declares textures with partial mipchains incomplete, and I'm not sure how a WebGL1-on-top-of-GLES2 implementation would handle this.

Reply:
Sorry, I forgot about GLES2, we need all levels.

If something isn't said in WebGL 1.0 then it's usually implied to follow GLES2.

Reply:
On the other hand, a texture can refer to a sampler object that uses minification without mip maps. Should we allow KTX2 files to have only level 0 in this case?

Reply:
Please review the updated language.

Reply:
LGTM

Reply:
Looks good to me.